frontend/src/metabase/visualizer/components/Visualizer/Visualizer.tsx

```diff
@@ -53,6 +54,7 @@ export const Visualizer = (props: VisualizerProps) => {
   const { className, onSave, saveLabel } = props;
   const { canUndo, canRedo, undo, redo } = useVisualizerHistory();
 
+  const dashboardId = useSelector(getDashboard)?.id;
```

Comment:
I'd suggest moving the selector closer to where it's used, i.e. to data importer's search results component